src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,369 @@
+                    leftNullable || rightNullable, diagnostics);
+            }
+
+            return BindTupleBinaryOperatorSingleInfo(node, kind, left, right, diagnostics);
+        }
+
+        private TupleBinaryOperatorInfo BindTupleBinaryOperatorSingleInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
+        {
+            TypeSymbol leftType = left.Type;
+            TypeSymbol rightType = right.Type;
+
+            if ((object)leftType != null && leftType.IsDynamic() || (object)rightType != null && rightType.IsDynamic())
+            {
+                return BindTupleDynamicBinaryOperatorSingleInfo(node, kind, left, right, diagnostics);
+            }
+
+            LookupResultKind resultKind;
+            BinaryOperatorAnalysisResult best = this.BinaryOperatorOverloadResolution(kind, left, right, node, diagnostics, out resultKind, out var _);
+
+            MethodSymbol resultMethod = null;
+            BinaryOperatorKind resultOperatorKind;
+            TypeSymbol convertedLeftType = null;
+            TypeSymbol convertedRightType = null;
+            TypeSymbol returnType;
+            bool hasErrors;
+
+            if (!best.HasValue)
+            {
+                resultOperatorKind = kind;
+                convertedLeftType = convertedLeftType ?? leftType ?? CreateErrorType();
+                convertedRightType = convertedRightType ?? rightType ?? CreateErrorType();
+                returnType = CreateErrorType();
+                hasErrors = true;
+            }
+            else
+            {
+                bool leftNull = left.IsLiteralNull();
+                bool rightNull = right.IsLiteralNull();
+
+                var signature = best.Signature;
+
+                bool isObjectEquality = signature.Kind == BinaryOperatorKind.ObjectEqual || signature.Kind == BinaryOperatorKind.ObjectNotEqual;
+
+                resultOperatorKind = signature.Kind;
+                resultMethod = signature.Method;
+                convertedLeftType = signature.LeftType;
+                convertedRightType = signature.RightType;
+                returnType = signature.ReturnType; // PROTOTYPE(tuple-equality) Do we need a special case for nullable equality? (see BindSimpleBinaryOperator)
+
+                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                hasErrors = isObjectEquality && !BuiltInOperators.IsValidObjectEquality(Conversions, leftType, leftNull, rightType, rightNull, ref useSiteDiagnostics);
+                diagnostics.Add(node, useSiteDiagnostics);
+            }
+
+            if (hasErrors)
+            {
+                ReportBinaryOperatorError(node, diagnostics, node.OperatorToken, left, right, resultKind);
+                //resultOperatorKind &= ~BinaryOperatorKind.TypeMask; // PROTOTYPE(tuple-equality) Not sure what this is for
```

Comment:
>Not sure what this is for [](start = 99, length = 25)

This removes the part of the operator kind that stores the bits designating the kind/type (e.g. enum, int, etc). There may not be any bits in the tuple case.

Reply:
Given that this method is for binding an element-wise comparison, it seems like this would make sense. But I found not observable effect of doing so. 
I'll just delete this note.

---
In reply to: [168057524](https://github.com/dotnet/roslyn/pull/23356#discussion_r168057524) [](ancestors = 168057524)